projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php

```diff
@@ -159,42 +193,46 @@ function callback( $path = '', $blog_id = 0 ) {
 			'fields'         => 'ids',
 		);
 
-		if ( ! is_user_logged_in () ) {
+		if ( ! is_user_logged_in() ) {
 			$query['has_password'] = false;
 		}
 
 		if ( isset( $args['meta_key'] ) ) {
 			$show = false;
-			if ( WPCOM_JSON_API_Metadata::is_public( $args['meta_key'] ) )
+			if ( WPCOM_JSON_API_Metadata::is_public( $args['meta_key'] ) ) {
 				$show = true;
-			if ( current_user_can( 'edit_post_meta', $query['post_type'], $args['meta_key'] ) )
+			}
+			if ( current_user_can( 'edit_post_meta', $query['post_type'], $args['meta_key'] ) ) {
 				$show = true;
+			}
 
-			if ( is_protected_meta( $args['meta_key'], 'post' ) && ! $show )
+			if ( is_protected_meta( $args['meta_key'], 'post' ) && ! $show ) {
 				return new WP_Error( 'invalid_meta_key', 'Invalid meta key', 404 );
+			}
 
 			$meta = array( 'key' => $args['meta_key'] );
-			if ( isset( $args['meta_value'] ) )
+			if ( isset( $args['meta_value'] ) ) {
 				$meta['value'] = $args['meta_value'];
+			}
 
 			$query['meta_query'] = array( $meta );
 		}
 
 		if (
 			isset( $args['sticky'] )
 		&&
-			( $sticky = get_option( 'sticky_posts' ) )
+			( $sticky = get_option( 'sticky_posts' ) ) // phpcs:ignore WordPress.CodeAnalysis.AssignmentInCondition.Found, Squiz.PHP.DisallowMultipleAssignments.FoundInControlStructure
```

Comment:
Do you think you could move the assignment before the condition?

Reply:
Let me know if dc5fb9e055537b69460d5318b4c5e7b919da778c works